common/utils/utils.go

```diff
@@ -506,6 +506,60 @@ func ReadSpannerSchema(ctx context.Context, conv *internal.Conv, client *sp.Clie
 	return nil
 }
 
+// CompareSchema compares the spanner schema of two conv objects and returns specific error if they don't match
+func CompareSchema(conv1, conv2 *internal.Conv) error {
+	if conv1.TargetDb != conv2.TargetDb {
+		return fmt.Errorf("target db don't match")
+	}
+	for t := range conv1.SpSchema {
+		sessionTable := conv1.SpSchema[t]
+		spannerTable := conv2.SpSchema[t]
+		if sessionTable.Name != spannerTable.Name || sessionTable.Parent != spannerTable.Parent ||
+			len(sessionTable.Pks) != len(spannerTable.Pks) || len(sessionTable.ColDefs) != len(spannerTable.ColDefs) ||
+			len(sessionTable.Indexes) != len(spannerTable.Indexes) {
+			return fmt.Errorf("table detail for table %v don't match", sessionTable.Name)
+		}
+		for i := range sessionTable.Pks {
+			if sessionTable.Pks[i].Col != spannerTable.Pks[i].Col || sessionTable.Pks[i].Desc != spannerTable.Pks[i].Desc {
+				return fmt.Errorf("primary keys for table %v don't match", sessionTable.Name)
+			}
+		}
+		for col := range sessionTable.ColDefs {
+			colDef := sessionTable.ColDefs[col]
+			spannerCol := spannerTable.ColDefs[col]
+			if colDef.Name != spannerCol.Name || colDef.NotNull != spannerCol.NotNull ||
+				colDef.T.IsArray != spannerCol.T.IsArray || colDef.T.Len != spannerCol.T.Len || colDef.T.Name != spannerCol.T.Name {
+				return fmt.Errorf("column detail for table %v don't match", sessionTable.Name)
+			}
+		}
+		for i := range sessionTable.Indexes {
+			sessionDbIndex := sessionTable.Indexes[i]
+			found := 0
+			for j := range spannerTable.Indexes {
```

Comment:
I am not sure if it is a `golang` thing, but can we use replace existing variables `t, i, j` with something more descriptive? 
It is hard to read the code where they are being referred to at once.